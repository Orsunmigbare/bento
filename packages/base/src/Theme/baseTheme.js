import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
