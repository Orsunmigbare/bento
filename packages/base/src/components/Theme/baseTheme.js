export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
