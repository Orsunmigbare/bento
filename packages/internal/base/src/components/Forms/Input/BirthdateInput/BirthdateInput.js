import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Input from "../Input"
import { applyVariants } from "../../../../theming"
import { themePropTypes } from "../../../../constants/themeContants"

class BirthdateInput extends React.Component {
  constructor(props) {
    super(props)
    // create a ref to store the textInput DOM element
    this.dayInput = React.createRef()
    this.handleChangeDayInput = this.handleChangeDayInput.bind(this)

    this.monthInput = React.createRef()
    this.handleChangeMonthInput = this.handleChangeMonthInput.bind(this)

    this.yearInput = React.createRef()
    this.handleChangeYearInput = this.handleChangeYearInput.bind(this)

    this.createDate = this.createDate.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.handleComplete = this.handleComplete.bind(this)
  }

  handleChangeDayInput(value) {
    this.handleChange()
    if (value.toString().length > 1) {
      this.monthInput.current.input.current.focus()

      if (
        this.monthInput.current.input.current.value.toString().length > 1 &&
        this.yearInput.current.input.current.value.toString().length > 1
      ) {
        this.handleComplete()
      } else {
        this.monthInput.current.input.current.focus()
      }
    }
  }

  handleChangeMonthInput(value) {
    this.handleChange()
    if (value.toString().length > 1) {
      if (
        this.dayInput.current.input.current.value.toString().length > 1 &&
        this.yearInput.current.input.current.value.toString().length > 1
      ) {
        this.handleComplete()
      } else {
        this.yearInput.current.input.current.focus()
      }
    }
  }

  handleChangeYearInput(value) {
    this.handleChange()
    if (value.toString().length > 3) {
      if (this.dayInput.current.input.current.value.toString().length < 1) {
        this.dayInput.current.input.current.focus()
      } else if (
        this.monthInput.current.input.current.value.toString().length < 1
      ) {
        this.monthInput.current.input.current.focus()
      } else {
        this.handleComplete()
      }
    }
  }

  createDate(d, m, y) {
    return new Date(y.toString() + "-" + m.toString() + "-" + d.toString())
  }

  handleChange() {
    if (this.props.onChange) {
      this.props.onChange(
        this.dayInput.current.input.current.value.toString() +
          this.monthInput.current.input.current.value.toString() +
          this.yearInput.current.input.current.value.toString()
      )
    }
  }

  handleComplete() {
    this.props.onComplete &&
      this.props.onComplete(
        this.createDate(
          this.dayInput.current.input.current.value,
          this.monthInput.current.input.current.value,
          this.yearInput.current.input.current.value
        )
      )
  }

  render() {
    const { id, name, labels } = this.props
    return (
      <Container variant={this.props.variant}>
        <Input
          label={labels[0] && labels[0]}
          placeholder="dd"
          id={`${id}Day`}
          name={`${name}Day`}
          type={"tel"}
          ref={this.dayInput}
          onChange={e => this.handleChangeDayInput(e.target.value)}
          variant={this.props.variant}
        />
        <Input
          label={labels[1] && labels[1]}
          placeholder="mm"
          id={`${id}Month`}
          name={`${name}Month`}
          type={"tel"}
          ref={this.monthInput}
          onChange={e => this.handleChangeMonthInput(e.target.value)}
          variant={this.props.variant}
        />
        <Input
          label={labels[2] && labels[2]}
          placeholder="yyyy"
          id={`${id}Year`}
          name={`${name}Year`}
          type={"tel"}
          ref={this.yearInput}
          onChange={e => this.handleChangeYearInput(e.target.value)}
          variant={this.props.variant}
        />
      </Container>
    )
  }
}

BirthdateInput.themeProps = {
  container: {
    name: "BIRTHDATE_INPUT_CONTAINER_STYLE",
    description: "Container style",
    type: themePropTypes.style
  }
}

const Container = styled.div`
  display: flex;
  ${applyVariants(BirthdateInput.themeProps.container)};
`

BirthdateInput.propTypes = {
  variant: PropTypes.string,
  onChange: PropTypes.func,
  onComplete: PropTypes.func,
  ids: PropTypes.array.isRequired,
  labels: PropTypes.array
}

BirthdateInput.defaultProps = {}

export default BirthdateInput
