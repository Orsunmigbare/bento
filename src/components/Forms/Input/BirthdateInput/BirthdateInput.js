import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Label from "../../Label/Label"
import Input from "../Input"
import { applyVariants } from "../../../../theming"
import { themePropTypes } from "../../../../constants/themeContants"

class BirthdateInput extends React.Component {
  constructor(props) {
    super(props)
    // create a ref to store the textInput DOM element
    this.dayInput = React.createRef()
    this.handleChangeDayInput = this.handleChangeDayInput.bind(this)

    this.monthInput = React.createRef()
    this.handleChangeMonthInput = this.handleChangeMonthInput.bind(this)

    this.yearInput = React.createRef()
    this.handleChangeYearInput = this.handleChangeYearInput.bind(this)

    this.createDate = this.createDate.bind(this)
    this.handleChange = this.handleChange.bind(this)
    this.handleComplete = this.handleComplete.bind(this)
  }

  handleChangeDayInput(value) {
    this.handleChange()
    if (value.toString().length > 1) {
      this.monthInput.current.focus()
      if (
        this.monthInput.current.value.toString().length > 1 &&
        this.yearInput.current.value.toString().length > 1
      ) {
        this.handleComplete()
      } else {
        this.monthInput.current.focus()
      }
    }
  }

  handleChangeMonthInput(value) {
    this.handleChange()
    if (value.toString().length > 1) {
      if (
        this.dayInput.current.value.toString().length > 1 &&
        this.yearInput.current.value.toString().length > 1
      ) {
        this.handleComplete()
      } else {
        this.yearInput.current.focus()
      }
    }
  }

  handleChangeYearInput(value) {
    this.handleChange()
    if (value.toString().length > 3) {
      if (this.dayInput.current.value.toString().length < 1) {
        this.dayInput.current.focus()
      } else if (this.monthInput.current.value.toString().length < 1) {
        this.monthInput.current.current.focus()
      } else {
        this.handleComplete()
      }
    }
  }

  createDate(d, m, y) {
    console.log(
      new Date(y.toString() + "-" + m.toString() + "-" + d.toString())
    )
    return new Date(y.toString() + "-" + m.toString() + "-" + d.toString())
  }

  handleChange() {
    if (this.props.onChange) {
      this.props.onChange(
        this.dayInput.current.value.toString() +
          this.monthInput.current.value.toString() +
          this.yearInput.current.value.toString()
      )
    }
  }

  handleComplete() {
    this.createDate(
      this.dayInput.current.value,
      this.monthInput.current.value,
      this.yearInput.current.value
    )
  }

  render() {
    const {
      id,
      name,
      labels,
      variant,
      className,
      autoFocus,
      disabled,
      onKeyDown,
      style,
      pattern,
      label,
      onChange,
      onComplete,
      ids,
      names,
      ...rest
    } = this.props
    console.log({ ...rest })
    return (
      <Outer variant={variant} className={className} style={style}>
        {label && <Label id={id}>{label}</Label>}
        <Container variant={variant}>
          <Input
            label={labels && labels[0] && labels[0]}
            placeholder="dd"
            id={ids ? (ids[0] ? ids[0] : `${id}Day`) : `${id}Day`}
            name={names ? (names[0] ? names[0] : `${name}Day`) : `${name}Day`}
            type="tel"
            ref={this.dayInput}
            onChange={e => this.handleChangeDayInput(e.target.value)}
            variant={variant}
            autoFocus={autoFocus}
            disabled={disabled}
            onKeyDown={onKeyDown}
            pattern={pattern}
            aria-labelledby={label && id}
          />
          <Input
            label={labels && labels[1] && labels[1]}
            placeholder="mm"
            id={ids ? (ids[1] ? ids[1] : `${id}Month`) : `${id}Month`}
            name={`${name}Month`}
            type="tel"
            ref={this.monthInput}
            onChange={e => this.handleChangeMonthInput(e.target.value)}
            variant={variant}
            disabled={disabled}
            onKeyDown={onKeyDown}
            pattern={pattern}
            aria-labelledby={label && id}
          />
          <Input
            label={labels && labels[2] && labels[2]}
            placeholder="yyyy"
            id={ids ? (ids[2] ? ids[2] : `${id}Year`) : `${id}Year`}
            name={`${name}Year`}
            type="tel"
            ref={this.yearInput}
            onChange={e => this.handleChangeYearInput(e.target.value)}
            variant={variant}
            disabled={disabled}
            onKeyDown={onKeyDown}
            pattern={pattern}
            aria-labelledby={label && id}
          />
        </Container>
      </Outer>
    )
  }
}

BirthdateInput.themeProps = {
  outer: {
    name: "BIRTHDATE_INPUT_OUTER_STYLE",
    description: "Outer style",
    type: themePropTypes.style
  },
  container: {
    name: "BIRTHDATE_INPUT_CONTAINER_STYLE",
    description: "Container style",
    type: themePropTypes.style
  }
}

const Outer = styled.div`
  ${applyVariants(BirthdateInput.themeProps.outer)};
`

const Container = styled.div`
  display: flex;
  ${applyVariants(BirthdateInput.themeProps.container)};
`

BirthdateInput.propTypes = {
  variant: PropTypes.string,
  onChange: PropTypes.func,
  onComplete: PropTypes.func,
  ids: PropTypes.array.isRequired,
  labels: PropTypes.array
}

BirthdateInput.defaultProps = {}

export default BirthdateInput
