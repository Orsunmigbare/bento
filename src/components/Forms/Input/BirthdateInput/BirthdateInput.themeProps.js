import { themePropTypes } from "../../../../theming"

const themeProps = {
  outer: {
    name: "BIRTHDATE_INPUT_OUTER_STYLE",
    description: "Outer style",
    type: themePropTypes.style
  },
  container: {
    name: "BIRTHDATE_INPUT_CONTAINER_STYLE",
    description: "Container style",
    type: themePropTypes.style
  }
}

export default themeProps
