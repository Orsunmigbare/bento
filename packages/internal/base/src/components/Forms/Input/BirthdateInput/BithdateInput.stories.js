import React from "react"
import { storiesOf } from "@storybook/react"
import BirthdateInput from "./BirthdateInput"

storiesOf("components|base/Forms/Input/BirthDateInput", module).add(
  "no props",
  () => {
    return <BirthdateInput label={"BithdateInput"} ids="9852145" />
  }
)
