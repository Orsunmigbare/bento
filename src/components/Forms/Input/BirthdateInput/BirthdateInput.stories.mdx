import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import BirthdateInput from "./BirthdateInput"

<Meta title="Components|Forms/Input/BirthdateInput" component={BirthdateInput}/>


# BirthdateInput
This components lets the user write inn their birthdate in seperated fields. The component also take a label prop, that sets
the label for the input field


```javascript
import {BirthdateInput} from "@staccx/bento"
```



## Standard BirthdateInput
The components creates three input fields, day, mont, year. 


<Preview>
  <Story name="Standard">
   <BirthdateInput label={"BirthdateInput"} ids={["Day","Month","Year"]} labels={["Day","Month","Year"]}/>
  </Story>
</Preview>





## Props for BirthdateInput
<Props of={BirthdateInput} />
