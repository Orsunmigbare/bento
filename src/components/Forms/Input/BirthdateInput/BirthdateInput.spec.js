import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import BirthdateInput from "./BirthdateInput"
import "@testing-library/jest-dom"

const setup = () => {
  const utils = render(
    <ThemeProvider theme={baseTheme}>
      <BirthdateInput
        labels={["Day", "Month", "Year"]}
        ids={["Day1", "Month1", "Year1"]}
        onChange={console.log}
      />
    </ThemeProvider>
  )
  const day = utils.getByLabelText("Day")
  const month = utils.getByLabelText("Month")
  const year = utils.getByLabelText("Year")
  return {
    day,
    month,
    year,
    ...utils
  }
}

describe("BirthdateInput", () => {
  describe("Snapshots", () => {
    it("All props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <BirthdateInput
              label="BirthdateInput"
              ids={["Day1", "Month1", "Year1"]}
              labels={["Day", "Month", "Year"]}
              onChange={console.log}
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("simulate input", () => {
      const { day, month, year } = setup()
      fireEvent.change(day, { target: { value: 12 } })
      fireEvent.change(month, { target: { value: 1 } })
      fireEvent.change(year, { target: { value: 1995 } })
      expect(day.value).toBe("12")
      expect(month.value).toBe("1")
      expect(year.value).toBe("1995")
    })
    it("simulate input2", () => {
      const { day, month, year } = setup()
      fireEvent.change(day, { target: { value: 12 } })
      fireEvent.change(month, { target: { value: 12 } })
      fireEvent.change(year, { target: { value: 1995 } })
      expect(day.value).toBe("12")
      expect(month.value).toBe("12")
      expect(year.value).toBe("1995")
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <BirthdateInput
          data-testid="testID"
          label="BirthdateInput"
          ids={["Day1", "Month1", "Year1"]}
          labels={["Day", "Month", "Year"]}
          onChange={console.log}
        />
      </ThemeProvider>
    )
    expect(screen.getByPlaceholderText("dd")).toBeInTheDocument()
  })
})
