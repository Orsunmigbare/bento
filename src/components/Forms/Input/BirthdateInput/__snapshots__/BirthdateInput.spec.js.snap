// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdateInput Snapshots All props 1`] = `
<div
  className="sc-AykKK gngxPY"
>
  <label
    className="sc-AykKC bonmue"
  >
    BirthdateInput
  </label>
  <div
    className="sc-fzXfLO fSfKbe"
  >
    <div
      className="sc-AykKI iDYrmt"
    >
      <label
        className="sc-AykKC bonmue sc-AykKH hciGUB"
        htmlFor="Day1"
      >
        Day
        
      </label>
      
      <input
        autoFocus={null}
        className="sc-AykKJ gjiytr"
        disabled={false}
        hidden={false}
        id="Day1"
        name="undefinedDay"
        onBlur={null}
        onChange={[Function]}
        onKeyDown={null}
        placeholder="dd"
        style={null}
        type="tel"
        value=""
      />
    </div>
    <div
      className="sc-AykKI iDYrmt"
    >
      <label
        className="sc-AykKC bonmue sc-AykKH hciGUB"
        htmlFor="Month1"
      >
        Month
        
      </label>
      
      <input
        autoFocus={null}
        className="sc-AykKJ gjiytr"
        disabled={false}
        hidden={false}
        id="Month1"
        name="undefinedMonth"
        onBlur={null}
        onChange={[Function]}
        onKeyDown={null}
        placeholder="mm"
        style={null}
        type="tel"
        value=""
      />
    </div>
    <div
      className="sc-AykKI iDYrmt"
    >
      <label
        className="sc-AykKC bonmue sc-AykKH hciGUB"
        htmlFor="Year1"
      >
        Year
        
      </label>
      
      <input
        autoFocus={null}
        className="sc-AykKJ gjiytr"
        disabled={false}
        hidden={false}
        id="Year1"
        name="undefinedYear"
        onBlur={null}
        onChange={[Function]}
        onKeyDown={null}
        placeholder="yyyy"
        style={null}
        type="tel"
        value=""
      />
    </div>
  </div>
</div>
`;
